### PR TITLE
tiltfile: send analytics for extensions [ch1142]

### PR DESCRIPTION
**Pull request by maiamcc**
this feels kind of weird but is the best implementation i could think of that matches the generic-ness of `LoadInterceptors` (i.e. if in the future we have other different kinds of `LoadInterceptors` that want to send data back to analytics, we can pretty easily)

alternately i could
- wire the `analytics` object through to the Extension itself and record the info at the point at which we parse the extension path, which seems awful, or
- store the relevant info on the Extension and type check the `LoadInterceptors` to see if any are a `tiltextensions.Extension`, and THEN pull the data off, which seems counter to the pretty generic way `LoadInterceptors` are build right now (of course I don't precisely understand WHY the current generic-ness, so... ¯\_(ツ)_/¯ )
- ??

### CODE REVIEW

File: internal/tiltfile/starkit/load_interceptor.go [hunk 1]
```diff
@@ -11,4 +11,7 @@ type LoadInterceptor interface {
 	// Ensure the content is present in the path returned
 	// Returns "" if this interceptor doesn't act on this path
 	LocalPath(t *starlark.Thread, path string) (string, error)
+
+	// Optionally, a LoadInterceptor can record some info to send to analytics later
+	AnalyticsInfo() []AnalyticsInfo
```
Comment (nicks) [inline internal/tiltfile/starkit/load_interceptor.go:16]: ya, this feels like a lot of layers of indirection

have you considered using the existing data model pattern for collecting this data?
https://github.com/tilt-dev/tilt/blob/master/internal/tiltfile/starkit/model.go

we have pretty good conventions for adding things to that data model and pulling it off at the end of execution in ways that prevent common mistakes.

**Review by nicks — COMMENTED** [review]
left a few high level comments!

File: internal/tiltfile/tiltfile_state.go [hunk 2]
```diff
@@ -205,7 +202,7 @@ func (s *tiltfileState) loadManifests(absFilename string, userConfigState model.
 		shlex.NewExtension(),
 		watch.NewExtension(),
 		loaddynamic.NewExtension(),
-		tiltextension.NewExtension(fetcher, tiltextension.NewLocalStore(filepath.Dir(absFilename))),
```
[the remark this answers is not in the document]
Comment (maiamcc) [reply internal/tiltfile/tiltfile_state.go:208]: ah, I just made it so i could dependency inject a fake version into unit tests, that's a super fair point and we should probably not do that.

File: internal/tiltfile/tiltextension/extension.go [hunk 3]
```diff
@@ -66,8 +85,15 @@ func (e *Extension) LocalPath(t *starlark.Thread, arg string) (string, error) {
 		_ = e.fetcher.CleanUp()
 	}()
 
+	// Record that we loaded this extension
+	e.info.ExtensionsLoaded[moduleName] = true
```
[the remark this answers is not in the document]
Comment (maiamcc) [reply internal/tiltfile/tiltextension/extension.go:89]: whoops! put the line in the wrong place.

File: internal/tiltfile/starkit/load_interceptor.go (hunk 1 above)
Comment (maiamcc) [reply internal/tiltfile/starkit/load_interceptor.go:16]: i honestly completely forgot about that part of tiltfile world. is https://github.com/tilt-dev/tilt/pull/4229 more like it? (def feels much better to me)

Comment (nicks) [reply internal/tiltfile/starkit/load_interceptor.go:16]: :+1:

**Pull request closed without merging**